hbase-server/src/test/java/org/apache/hadoop/hbase/util/TestHBaseFsck.java

```diff
@@ -882,6 +882,80 @@ public void testDupeStartKey() throws Exception {
       assertNoErrors(hbck2);
       assertEquals(0, hbck2.getOverlapGroups(table).size());
       assertEquals(ROWKEYS.length, countRows());
+
+      MiniHBaseCluster cluster = TEST_UTIL.getHBaseCluster();
+      long totalRegions = cluster.countServedRegions();
+
+      // stop a region servers and run fsck again
+      cluster.stopRegionServer(server);
+      cluster.waitForRegionServerToStop(server, 60);
+
+      // wait for all regions to come online.
+      while (cluster.countServedRegions() < totalRegions) {
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException e) {}
```

Comment:
does this `InterruptedException` need to be caught? Can't the test method throw it?

Reply:
Sleep can be waken up by interruption before time is up, that should be normal, I want it not to break the loop until we finished waiting for all regions to come up.
Thread.sleep() method is defined this way: https://www.tutorialspoint.com/java/lang/thread_sleep_millis.htm

Reply:
This pattern is often considered a code smell. The InterruptedException is not handled and so the thread's interrupt status is swallowed. You would want to at least do:

        try {
          Thread.sleep(100);
        } catch (InterruptedException e) {
          Thread.currentThread().interrupt(); // restore interrupt status
        }
        
But given this is test code this is trying to hard. I would just let the test method throw the exception FWIW